src/index/index.c

```diff
@@ -206,10 +239,9 @@ void Index_RemoveField
 
 	uint fields_count = array_len(idx->fields);
 	for(uint i = 0; i < fields_count; i++) {
-		if(idx->fields_ids[i] == attribute_id) {
-			rm_free(idx->fields[i]);
+		if(idx->fields[i].id == attribute_id) {
+			rm_free(idx->fields[i].name);
```

Comment:
Same as we have an `IndexField` constructor we should have a destructor.
`IndexField_Free`

Reply:
I believe we forgot:
`if(idx->fields[i].phonetic) rm_free(idx->fields[i].phonetic);`
For this reason I believe we need a destructor